programs/uxd/src/instructions/redeem.rs

```diff
@@ -160,24 +167,67 @@ struct DepositoryInfoForRedeem<'info> {
 
 pub(crate) fn handler(ctx: Context<Redeem>, redeemable_amount: u64) -> Result<()> {
     // Gather all the onchain states we need (caps, weights and supplies)
-    let controller = ctx.accounts.controller.load()?;
+    let mut controller = ctx.accounts.controller.load_mut()?;
     let identity_depository = ctx.accounts.identity_depository.load()?;
     let mercurial_vault_depository = ctx.accounts.mercurial_vault_depository.load()?;
     let credix_lp_depository = ctx.accounts.credix_lp_depository.load()?;
 
+    // Compute max outflow per epoch (max of bps/amount options)
+    let current_outflow_limit_per_epoch_amount = std::cmp::max(
+        controller.outflow_limit_per_epoch_amount.into(),
+        checked_div(
+            checked_mul(
+                controller.redeemable_circulating_supply,
+                controller.outflow_limit_per_epoch_bps.into(),
+            )?,
+            BPS_POWER.into(),
+        )?,
+    );
+
+    // How long ago was the last outflow
+    let now_timestamp = Clock::get()?.unix_timestamp;
+    let last_outflow_seconds_ago = checked_sub(now_timestamp, controller.last_outflow_timestamp)?;
+
+    // How much was unlocked by waiting since last redeem
+    let unlocked_outflow_amount = checked_as_u64(checked_div(
+        checked_mul(
+            checked_as_u128(last_outflow_seconds_ago)?,
+            current_outflow_limit_per_epoch_amount,
+        )?,
+        controller.seconds_per_epoch.into(),
+    )?)?;
+
+    // How much outflow in the current epoch right before the redeem IX
+    let previous_epoch_outflow_amount = if controller.epoch_outflow_amount > unlocked_outflow_amount
+    {
+        checked_sub(controller.epoch_outflow_amount, unlocked_outflow_amount)?
```

Comment:
same

Reply:
I'm confused, I think I should leae it as-is right ? based on your comment above: https://github.com/UXDProtocol/uxd-program/pull/288#discussion_r1285332840